### PR TITLE
Fixed "Attempted to get trait from destroyed object" in *OrderGenerator.SelectionChanged 

### PR DESCRIPTION
**Pull request by Mailaender**
This is a fix for https://github.com/Dzierzan/OpenSA/issues/134.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/AttackMove.cs [hunk 1]
```diff
@@ -123,7 +123,7 @@ protected virtual IEnumerable<Order> OrderInner(World world, CPos cell, MouseInp
 
 		public override void SelectionChanged(World world, IEnumerable<Actor> selected)
 		{
-			subjects = selected.SelectMany(a => a.TraitsImplementing<AttackMove>()
+			subjects = selected.Where(s => !s.IsDead).SelectMany(a => a.TraitsImplementing<AttackMove>()
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/AttackMove.cs:126]: @teinarss (or anyone else that knows), how does this added `Where` stack up in terms of perf vs. just putting that check as first check in the `SelectMany`?

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/AttackMove.cs:126]: I guess performance wise it is better to sort out the dead before doing a trait lookup (which crashes the game) plus I am not sure if it is syntactically possible to do a `SelectMany` which includes a binary check.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/AttackMove.cs:126]: Something like
```
subjects = selected.SelectMany(a =>
{
	if (!a.IsDead)
		return a.TraitsImplementing<AttackMove>().Select(am => new TraitPair<AttackMove>(a, am));

	return Enumerable.Empty<TraitPair<AttackMove>>();
}).ToArray();
```
should work, but I don't think it is really worth it here.

**Review by abcdefg30 — APPROVED** [review]

**Review by phrohdoh — APPROVED** [review]
Code changes LGTM.  Have not tested in-game.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/610fcb58087b4565a09266b0eb4e08a90835b74d...fabe96982bd1ef4d070a72fcecf8a9079706fd3f)

**Pull request merged**